language/move-lang/src/typing/expand.rs

```diff
@@ -0,0 +1,327 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use super::core::{self, Context};
+use crate::{
+    naming::ast::{
+        BaseType, BaseType_, FunctionSignature, SingleType, SingleType_, TParam, Type, Type_,
+    },
+    parser::ast::{Kind, Kind_},
+    shared::*,
+    typing::ast as T,
+};
+use std::collections::BTreeSet;
+
+//**************************************************************************************************
+// Functions
+//**************************************************************************************************
+
+pub fn function_body_(context: &mut Context, b_: &mut T::FunctionBody_) {
+    match b_ {
+        T::FunctionBody_::Native => (),
+        T::FunctionBody_::Defined(es) => sequence(context, es),
+    }
+}
+
+pub fn function_signature(context: &mut Context, sig: &mut FunctionSignature) {
+    for (_, st) in &mut sig.parameters {
+        single_type(context, st);
+    }
+    type_(context, &mut sig.return_type)
+}
+
+pub fn function_acquires(context: &mut Context, bs: &mut BTreeSet<BaseType>) {
+    let old = std::mem::replace(bs, BTreeSet::new());
+    for mut bt in old {
+        base_type(context, &mut bt);
+        assert!(bs.insert(bt));
```

Comment:
Shouldn't  this add to the errors for the acquires list? I can't see anywhere else where the acquires list is checked for uniqueness.

Reply:
Its already a set

Reply:
And uniqueness is checked in naming